src/arch/src/aarch64/fdt.rs

```diff
@@ -306,23 +311,51 @@ fn create_cpu_nodes(fdt: &mut Vec<u8>, vcpu_mpidr: &[u64]) -> Result<()> {
     append_property_u32(fdt, "#address-cells", 0x02)?;
     append_property_u32(fdt, "#size-cells", 0x0)?;
 
-    let num_cpus = vcpu_mpidr.len();
     for (cpu_index, mpidr) in vcpu_mpidr.iter().enumerate() {
         let cpu_name = format!("cpu@{:x}", cpu_index);
         append_begin_node(fdt, &cpu_name)?;
         append_property_string(fdt, "device_type", "cpu")?;
         append_property_string(fdt, "compatible", "arm,arm-v8")?;
-        if num_cpus > 1 {
-            // If the microVM has more than 1 vcpu we need to enable the power
-            // state coordination interface (PSCI) which will decide for us which
-            // vcpu is running or halted.
-            append_property_string(fdt, "enable-method", "psci")?;
-        }
+        // The power state coordination interface (PSCI) needs to be enabled for
+        // all vcpus.
+        append_property_string(fdt, "enable-method", "psci")?;
+        // For info on how to add cache information on arm, checkout
+        // https://github.com/devicetree-org/devicetree-specification/releases/download/v0.3/devicetree-specification-v0.3.pdf,
+        // section 3.8.
+        append_property_u32(fdt, "i-cache-size", 65536)?;
+        append_property_u32(fdt, "d-cache-size", 65536)?;
+        append_property_u32(fdt, "next-level-cache", CACHE2_PHANDLE)?;
+
         // Set the field to first 24 bits of the MPIDR - Multiprocessor Affinity Register.
         // See http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0488c/BABHBJCI.html.
         append_property_u64(fdt, "reg", mpidr & 0x7F_FFFF)?;
+
+        // Attach CPU to Numa0.
+        append_property_u32(fdt, "numa-node-id", 0)?;
         append_end_node(fdt)?;
     }
+
+    // Setting up L2 cache. The values copy the cache topology for an m6g machine.
```

Comment:
Can we read these values from the host instead of hard-coding them for m6g ?

Reply:
We need to either get the topology from the host or allow the user to set them up so this makes sense across multiple instance types with different topologies.